haystack/utils/deepsetcloud.py

```diff
@@ -967,3 +1112,169 @@ def get_file_client(
         """
         client = DeepsetCloudClient(api_key=api_key, api_endpoint=api_endpoint)
         return FileClient(client=client, workspace=workspace)
+
+
+class DeepsetCloudExperiments:
+    """
+    A facade to conduct and manage experiments within deepset Cloud.
+
+    To start a new experiment run:
+    1. Choose a pipeline to evaluate using `list_pipelines()`
+    2. Choose an evaluation set using `list_evaluation_sets()`
+    3. Create and start a new run using `create_and_start_run()`
+    4. Track the run using `get_run()`
+    """
+
+    @classmethod
+    def list_pipelines(
+        cls, workspace: str = "default", api_key: Optional[str] = None, api_endpoint: Optional[str] = None
+    ) -> List[dict]:
+        """
+        Lists all pipeline configs available on Deepset Cloud.
+
+        :param workspace: workspace in Deepset Cloud
+        :param api_key: Secret value of the API key.
+                        If not specified, will be read from DEEPSET_CLOUD_API_KEY environment variable.
+        :param api_endpoint: The URL of the Deepset Cloud API.
+                             If not specified, will be read from DEEPSET_CLOUD_API_ENDPOINT environment variable.
+
+        Returns:
+            list of dictionaries: List[dict]
+            each dictionary: {
+                        "name": str -> `pipeline_config_name` to be used in `load_from_deepset_cloud()`,
+                        "..." -> additional pipeline meta information
+                        }
+            example:
+                    [{'name': 'my_super_nice_pipeline_config',
+                        'pipeline_id': '2184e0c1-c6ec-40a1-9b28-5d2768e5efa2',
+                        'status': 'DEPLOYED',
+                        'created_at': '2022-02-01T09:57:03.803991+00:00',
+                        'deleted': False,
+                        'is_default': False,
+                        'indexing': {'status': 'IN_PROGRESS',
+                        'pending_file_count': 3,
+                        'total_file_count': 31}}]
+        """
+        client = DeepsetCloud.get_pipeline_client(api_key=api_key, api_endpoint=api_endpoint, workspace=workspace)
+        pipeline_config_infos = list(client.list_pipeline_configs())
+        return pipeline_config_infos
+
+    @classmethod
+    def list_evaluation_sets(
+        cls, workspace: str = "default", api_key: Optional[str] = None, api_endpoint: Optional[str] = None
+    ) -> List[dict]:
+        client = DeepsetCloud.get_evaluation_set_client(api_key=api_key, api_endpoint=api_endpoint, workspace=workspace)
+        return client.get_evaluation_sets()
+
+    @classmethod
+    def list_runs(
+        cls, workspace: str = "default", api_key: Optional[str] = None, api_endpoint: Optional[str] = None
+    ) -> List[dict]:
+        client = DeepsetCloud.get_eval_run_client(api_key=api_key, api_endpoint=api_endpoint, workspace=workspace)
+        return client.get_eval_runs()
```

Comment:
Regarding the names of the methods: In `DeepsetCloudExperiments` we have `list_runs` and `get_run`. They call `get_run` and `get_runs` in `EvaluationRunClient`. What about aligning this and using `get_runs` in both cases?

Reply:
Looking closer into that I found the reason why I called it `list` in the first place. There is also `list_evaluation_sets` and `list_pipelines`.  As the runs are the central piece of this API and you can act on (and not only list it) `get_runs` and `list_...` for the others might make sense. How do you feel about that?

Reply:
I understand the idea of having runs as the central piece of the API and I saw that we also have `list_pipeline_configs` and `get_pipeline_config`. In my opinion `get_...` would be more consistent than `list...` though. For example, if we have `list_evaluation_sets ` then why not also rename `get_labels` to `list_labels`? 
If `list_runs` or `list_pipelines` only returned a list of names then I would agree with having a method name that differs from `get_run` and `get_pipeline`. However, we actually return a list of runs or a list of pipelines (full dictionary with more than just the name) in the same way as `get_run` returns a single run. So I would prefer to name the methods more based on what they do (not how) rather than how we imagine them to be used. Maybe we add functionality that allows acting on evaluation sets later, who knows.

Reply:
Maybe @masci can share his opinion on that naming topic? Would be interested to hear a third opinion on this. 🙂

Reply:
I had a look at the module at large and I feel like this conversation might be a symptom of something bigger.

I would suggest at some point we stop and reassess the API consistency for these classes, and in the meantime we counterbalance naming issues with abundant documentation.